stores/listingCart.ts

```diff
@@ -0,0 +1,103 @@
+import { defineStore } from 'pinia'
+import { EntityWithId } from '~/components/rmrk/service/scheme'
+import { ComputedRef, set } from 'vue'
+import type { Prefix } from '@kodadot1/static'
+
+export type ListCartItem = {
+  id: string
+  name: string
+  urlPrefix: string
+  price: string
+  listPrice?: number
+  collection: EntityWithId
+}
+
+type ID = string
+
+interface State {
+  items: ListCartItem[]
+  owned: ListCartItem[]
+  chain: ComputedRef<Prefix>
+}
+
+const localStorage = useLocalStorage<ListCartItem[]>('listingCart', [])
+export const useListingCartStore = defineStore('listingCart', {
+  state: (): State => ({
+    items: localStorage.value,
+    owned: [],
+    chain: usePrefix().urlPrefix,
+  }),
+  getters: {
+    itemsInChain: (state): ListCartItem[] =>
+      state.items.filter((item) => item.urlPrefix === state.chain),
+    count() {
+      return this.itemsInChain.length
+    },
```

Comment:
```suggestion
    count: () => this.itemsInChain.length,
```

Reply:
this won't work since the scoping of the lexical this is different.
In this method, the this context is dynamically bound, meaning that when you call count, this refers to the object that owns the count method. 

Arrow functions don't bind their own this value; instead, they inherit it from the enclosing scope.
this.itemsInChain won't work

Reply:
This won't work. `this` has a different context here.